evaluation/tru_shared.py

```diff
@@ -0,0 +1,159 @@
+        return Tru()
+
+
+def getFeedbackFunctions(pipeline, golden_set):
+    # Initialize provider class
+    azureOpenAI = AzureOpenAI(deployment_name="gpt-35-turbo")
+
+    context = App.select_context(pipeline)
+
+    # Define a groundedness feedback function
+    grounded = Groundedness(groundedness_provider=azureOpenAI)
+    f_groundedness = (
+        Feedback(grounded.groundedness_measure_with_cot_reasons, name="groundedness")
+        .on(context.collect()).on_output()
+        .aggregate(grounded.grounded_statements_aggregator)
+    )
+
+    # Question/answer relevance between overall question and answer.
+    f_answer_relevance = (
+        Feedback(azureOpenAI.relevance_with_cot_reasons, name="answer_relevance")
+        .on_input_output()
+    )
+
+    # Question/statement relevance between question and each context chunk.
+    f_context_relevance = (
+        Feedback(azureOpenAI.qs_relevance_with_cot_reasons, name="context_relevance")
+        .on_input().on(context)
+        .aggregate(np.mean)
+    )
+
+    # GroundTruth for comparing the Answer to the Ground-Truth Answer
+    ground_truth_collection = GroundTruthAgreement(golden_set, provider=azureOpenAI)
+    f_answer_correctness = (
+        Feedback(ground_truth_collection.agreement_measure, name="answer_correctness")
+        .on_input_output()
+    )
+    return [f_answer_relevance, f_context_relevance, f_groundedness, f_answer_correctness]
+
+
+def getAzureChatModel(framework: Framework, deployment_name: str, model_version: str):
+    if framework == Framework.LANG_CHAIN:
+        return AzureChatOpenAI(
+            azure_deployment=deployment_name,
+            openai_api_version="2023-05-15",
+            model_version=model_version,
+            temperature=temperature,
+        )
+    elif framework == Framework.LLAMA_INDEX:
+        return LlamaAzureChatOpenAI(
+            deployment_name=deployment_name,
+            model=deployment_name,
+            api_key=os.getenv("AZURE_OPENAI_API_KEY"),
+            api_version="2023-05-15",
+            model_version=model_version,
+            temperature=temperature,
+        )
+    else:
+        raise Exception(f"Unknown framework: {framework} specified for getChatModel()")
+
+
+def getAzureEmbeddingsModel(framework: Framework):
+    if framework == Framework.LANG_CHAIN:
+        return AzureOpenAIEmbeddings(
+            azure_deployment="text-embedding-ada-002",
+            openai_api_version="2023-05-15"
+        )
+    elif framework == Framework.LLAMA_INDEX:
+        return AzureOpenAIEmbedding(
+            deployment_name="text-embedding-ada-002",
+            model="text-embedding-ada-002",
+            api_key=os.getenv("AZURE_OPENAI_API_KEY"),
+            api_version="2023-05-15",
+            temperature=temperature,
+        )
+    else:
+        raise Exception(f"Unknown framework: {framework} specified for getEmbeddingsModel()")
+
+
+def getAstraVectorStore(framework: Framework, collection_name: str):
```

Comment:
python convention is `get_astra_vector_store`: https://www.techversantinfotech.com/python-naming-conventions-points-you-should-know/

Reply:
updated all method names in `tru_shared.py`